src/posts/2025-09-16-kernel-basics-for-tetragon/index.md

```diff
@@ -0,0 +1,226 @@
+---
+path: '/blog/2025/09/16/kernel-basics-for-tetragon'
+date: '2025-09-16T12:00:00.000Z'
+title: 'Linux Kernel Fundamentals for Writing Effective Tetragon Tracing Policies '
+ogImage: cover.png
+isFeatured: false
+ogSummary: 'Learn the basics of the Linux Kernel needed to effectively write good Tetragon tracing policies'
+categories:
+  - Community
+tags:
+  - Tetragon
+---
+
+**_Author: Paul Arah, Isovalent@Cisco_**
+
+![](cover.png)
+
+When you write Tetragon tracing policies, you’re not writing arbitrary sets of rules; you're programming against the kernel execution path itself. Every policy you create hooks directly into kernel functions, intercepts system calls, and examines kernel data structures. This power comes with responsibility. Without an understanding of how the Linux kernel works, you'll find yourself writing policies that are ineffective, overly broad, or worse, missing the exact events you are trying to find.
+
+This blog is meant to be a pointer guide; it won’t make you a kernel hacker overnight, but it aims to cover some core Linux knowledge essential for crafting effective Tetragon tracing policies. We’ll connect kernel fundamentals such as user vs. kernel space, system calls, process structures, namespaces, and more to practical tracing policy examples. A basic familiarity with Linux and a base-level understanding of what Tetragon is will be enough to follow along.
```

Comment:
`This blog post`? I am not the best at writing english so if it's not correct do not consider my comment 😇